app/src/main/java/com/eveningoutpost/dexdrip/Models/blueReader.java

```diff
@@ -122,26 +165,39 @@ private static synchronized void processNewTransmitterData(TransmitterData trans
             PersistentStore.setLong("blueReader_Full_Battery", transmitterData.sensor_battery_level);
             Log.i(TAG, "blueReader_Full_Battery set to: " + transmitterData.sensor_battery_level) ;
         }
-        Pref.setInt("bridge_battery", ((transmitterData.sensor_battery_level - 3300) * 100 / (((int) (long) PersistentStore.getLong("blueReader_Full_Battery"))-3300)));
-        sensor.latest_battery_level = ((transmitterData.sensor_battery_level - 3300) * 100 / (((int) (long) PersistentStore.getLong("blueReader_Full_Battery"))-3300));
-        blueReaderDays = 6.129200670865791d / (1d + Math.pow(((double)transmitterData.sensor_battery_level/3763.700630306379d),(-61.04241888028577d)));
+        int localBridgeBattery =((transmitterData.sensor_battery_level - 3300) * 100 / (((int) (long) PersistentStore.getLong("blueReader_Full_Battery"))-3300));
+        Pref.setInt("bridge_battery", localBridgeBattery);
+        sensor.latest_battery_level = localBridgeBattery;
+        blueReaderDays = 6.129200670865791d / (1d + Math.pow(((double)transmitterData.sensor_battery_level/3763.700630306379d),(-61.04241888028577d))); //todo compare with test-formular, and new Data of batterylog
         if (transmitterData.sensor_battery_level < 3600) {
             blueReaderDays=blueReaderDays + 0.1d;
         }
         blueReaderDays = ((Math.round((blueReaderDays)*10d)/10d));
 
         PersistentStore.setString("bridge_battery_days", String.valueOf(blueReaderDays));
         sensor.save();
-
+        if (Pref.getBooleanDefaultFalse("blueReader_writebatterylog")) {
+            final String dir = getExternalDir();
+            makeSureDirectoryExists(dir);
+            writeLog(dir + BatLog,
+                    DateFormat.format("yyyyMMdd-kkmmss", System.currentTimeMillis()).toString() + "|" +
+                            PersistentStore.getLong("blueReader_Full_Battery") + "|" +
+                            transmitterData.sensor_battery_level + "|" +
+                            sensor.latest_battery_level + "|" +
+                            blueReaderDays
+            );
+        }
         DexCollectionService.last_transmitter_Data = transmitterData;
         Log.d(TAG, "BgReading.create: new BG reading at " + timestamp + " with a timestamp of " + transmitterData.timestamp);
         BgReading.create(transmitterData.raw_data, transmitterData.filtered_data, xdrip.getAppContext(), transmitterData.timestamp);
+
     }
 
 
     public static ByteBuffer initialize() {
-        Log.i(TAG, "initialize!");
-        Pref.setInt("bridge_battery", 0); //force battery to no-value before first reading
+        Log.i(TAG, "initialize blueReader!");
+        Pref.setInt("bridge_battery", 0);
+        PersistentStore.setDouble("blueReaderFirmwareValue", 0);
```

Comment:
is this covered by tests?

Reply:
Not for now, as this was new way, and implemented for future needs. will implement it.